### DISCUSSION TITLE
The debug info shows lots of lookups, even if I filter out stores

### DISCUSSION THREAD

**Discussion by liam-maps** [jef/streetmerchant#1675] [Help] [answered]
Hello to everyone.
First of all, thank @jef for this amazing piece of code.
I haven't scored any card yet, but I'm having a lot of fun tinkering with it.
The thing is that I feel the queries are quite slow among my selected stores and products.
From query to query of the same store and product it can pass up to one minute, or at least that's the time it takes to show the log on screen.
But, I've activated the debug logging level and I'm seeing a lot of "lookups" on all of the stores, even if I haven't selected it, for example:
[17:34:18] debug :: [bandh] Starting lookup...
[17:34:18] debug :: [bandh] Lookup done, next one in 9241.66014844086 ms
[17:34:18] debug :: [corsair] Starting lookup...
[17:34:18] debug :: [corsair] Lookup done, next one in 8425.470059827421 ms
[17:34:18] debug :: [proshop-dk] Starting lookup...
[17:34:18] debug :: [proshop-dk] Lookup done, next one in 5844.90320889416 ms

From time to time, among all of that debug messages some of my selected stores appear, but I have the feeling that I can economise on that queries and save bandwidth.
Is the script really querying all the available stores and products and then filtering it out from the output?

Cheers.

**Comment by jef** [comment, marked as answer]
Yeah, those have to do with link builders. They are actually being skipped, but the debug logs are still there. Probably should put those  logs within that if statement as well, but they still exist.

**Comment by liam-maps** [comment]
Thanks for your reply.
